src/main/java/com/treasuredata/bigdam/log/Log.java

```diff
@@ -298,17 +300,13 @@ private EventTime eventTime(final Instant now)
         return EventTime.fromEpoch((int) now.getEpochSecond(), now.getNano());
     }
 
-    private Map<String, Object> buildEvent(final String messageKey, final String message, final Throwable e, final Map<String, ? extends Object> attrs)
+    private Map<String, Object> filterAttrs(final Map<String, ? extends Object> attrs)
     {
-        HashMap<String, Object> event = new HashMap<>();
-        if (messageKey != null) {
-            event.put(messageKey, message);
-        }
-        if (e != null) {
-            event.put("errorClass", e.getClass().getName());
-            event.put("error", e.getMessage());
+        if (attrs == null) {
+            return ImmutableMap.of();
         }
-        event.putAll(defaultAttributes);
+
+        HashMap<String, Object> event = new HashMap<>();
```

Comment:
Could you add a comment why you put all entries to another Map?

Reply:
OK.